garmin_txt.cc

```diff
@@ -1096,13 +1081,13 @@ parse_waypoint()
     int i;
     double d;
     int field_no = header_fields[waypt_header][column];
-
     switch (field_no) {
     case  1:
-      wpt->shortname = DUPSTR(str);
+      wpt->shortname = (str);
       break;
     case  2:
-      wpt->notes = DUPSTR(str);
+      // Force a truncated string to prevent doubling elsewhere.
+      wpt->notes = str && str[0] == 0 ? 0 : str ;
```

Comment:
assigning 0 to a QString seems odd.
It is bizarre that this compiles but QString a; a=0; fails to compile with error: ambiguous overload for ‘operator=’.  I guess there is some implicit conversion with the ternary operator going on.

The intent seems to be to leave notes null instead of empty when str is a pointer to zero length C string, so a better comment might be "Force a null string to prevent ..."

This would hurt my brain less and works because wpt was just constructed with notes being null:
If (str && (str[0] != '\0')) {
  wpt->notes = str;
}